webapp/src/components/soundEffectEditor/SoundGallery.tsx

```diff
@@ -18,33 +18,109 @@ export interface SoundGalleryProps {
+                break;
+            case "ArrowRight":
+                playItemRefs.current[current].focus();
+                event.preventDefault();
+                break;
+            case "Space":
+            case "Enter":
+                selectItemRefs.current[current].click();
+                event.stopPropagation();
+                event.preventDefault();
+            default:
+        }
+    }
+
+    function previewNav(
+        prev: number,
+        next: number,
+        current: number,
+        event: React.KeyboardEvent<HTMLElement>) {
+        switch(event.code) {
+            case "ArrowDown":
+                playItemRefs.current[next].focus();
+                event.preventDefault();
+                break;
+            case "ArrowUp":
+                playItemRefs.current[prev].focus();
+                event.preventDefault();
+                break;
+            case "ArrowLeft":
+                selectItemRefs.current[current].focus();
+                event.preventDefault();
+                break;
+            default:
+        }
+    }
+
     return <div className={classList("sound-gallery", visible && "visible")} aria-hidden={!visible}>
         <div className="sound-gallery-scroller">
-            {sounds.map((item, index) =>
-                <div
-                    key={index}
-                    className="common-button"
-                    title={item.name}
-                    role="button"
-                    tabIndex={0}
-                    onKeyDown={fireClickOnEnter}
-                    onClick={() => onSoundSelected(item.sound)}>
-
-                    <SoundGalleryEntry {...item} useMixerSynthesizer={useMixerSynthesizer} />
-                </div>
-            )}
+            {sounds.map((item, index) => {
+                    const prev = (index + sounds.length - 1) % sounds.length;
+                    const next = (index + 1) % sounds.length;
+                    return(<div
+                        key={index}
+                        onClick={() => onSoundSelected(item.sound)}
+                        className="common-button">
+
+                        <SoundGalleryEntry
+                            {...item}
+                            useMixerSynthesizer={useMixerSynthesizer}
+                            onClick={() => onSoundSelected(item.sound)}
+
+                            playReference={ref => playItemRefs.current[index] = ref}
+                            selectReference={ref => selectItemRefs.current[index] = ref}
+
+                            previewKeyDown={evt => previewNav(prev, next, index, evt)}
+                            selectKeyDown={evt => selectNav(prev, next, index, evt)}
+                        />
+                    </div>);
+                })
+            }
         </div>
     </div>
 }
 
-
 const SoundGalleryEntry = (props: SoundGalleryItemProps) => {
-    const { sound, name, useMixerSynthesizer } = props;
+    const { 
```

Comment:
Remove trailing space
```suggestion
    const {
```